### PR TITLE
Removed c-ares

### LINKED ISSUE #5004: Android crash (ares_get_android_server_list)
**Describe the bug (REQUIRED)**
```
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
pid: 0, tid: 0 >>> com.potatojam.onet_paradise.match_two_tiles <<<

backtrace:
  #00  pc 00000000003338b6  /system/lib/libart.so (art::Thread::~Thread()+205)
  #00  pc 0000000000340291  /system/lib/libart.so (art::ThreadList::Unregister(art::Thread*)+1528)
  #00  pc 00000000003184df  /system/lib/libart.so (art::Runtime::DetachCurrentThread()+234)
  #00  pc 000000000023beb3  /system/lib/libart.so (art::JII::DetachCurrentThread(_JavaVM*)+42)
  #00  pc 00000000001d7bcc  /data/app/com.potatojam.onet_paradise.match_two_tiles-1/lib/arm/libOnetParadise.so (ares_get_android_server_list)
```
image

**To Reproduce (REQUIRED)**
Unfortunately I don't know how to reproduce this issue, I see this crash in Google Console.

**Expected behavior (REQUIRED)**

**Defold version (REQUIRED):**
  - Version 1.2.169

**Platforms (REQUIRED):**
 - Platforms: Android
 - OS: Android 6.0 and Android 7.0
 - Devices:
Samsung Galaxy Grand Prime Plus
BQru BQ-4072 Strike Mini
Samsung Galaxy On5
BQS-5020
Huawei 荣耀畅享7 Plus
BQru BQ-5594 Strike Power Max

**Additional context (OPTIONAL):**

### LINKED ISSUE #5775: Crash during engine shutdown while calling HTTP cache flush
**Describe the bug (REQUIRED)**
We have received reports of crashes during engine shutdown while dmHttpCache::Flush() is running

**Platforms (REQUIRED):**
 - Reported on iOS

```
Thread 0:
0   xxxxxxxxxxx                  	0x00000001050acba4 lj_gc_separateudata + 76
1   xxxxxxxxxxx                  	0x00000001050bbad4 lua_close + 232
2   xxxxxxxxxxx                  	0x00000001050bbad4 lua_close + 232
3   xxxxxxxxxxx                  	0x000000010508d0a4 dmScript::DeleteContext(dmScript::Context*) + 13832356 (script.cpp:92)
4   xxxxxxxxxxx                  	0x0000000104362dd8 dmEngine::Delete(dmEngine::Engine*) + 28120 (engine.cpp:332)
5   xxxxxxxxxxx                  	0x000000010516421c ShutdownEngine + 14713372 (AppDelegate.m:127)

Thread 2 Crashed:
0   xxxxxxxxxxx                  	0x000000010512215c dmHttpCache::Flush(dmHttpCache::Cache*) + 14442844 (http_cache.cpp:365)
1   xxxxxxxxxxx                  	0x000000010508c538 dmHttpService::Loop(void*) + 13829432 (http_service.cpp:333)
2   xxxxxxxxxxx                  	0x000000010508c538 dmHttpService::Loop(void*) + 13829432 (http_service.cpp:333)
3   xxxxxxxxxxx                  	0x000000010513a5cc dmThread::ThreadStartProxy(void*) + 14542284 (thread.cpp:43)
4   libsystem_pthread.dylib       	0x00000001d803bc9c _pthread_start + 288 (pthread.c:881)
5   libsystem_pthread.dylib       	0x00000001d8040880 thread_start + 8
```

Alternative:

```
Thread 3 Crashed:
0   libsystem_kernel.dylib        	0x00000001c6efd414 __pthread_kill + 8
1   libsystem_pthread.dylib       	0x00000001e4a58b50 pthread_kill + 272 (pthread.c:1392)
2   libsystem_c.dylib             	0x00000001a23dbbf4 __abort + 116 (abort.c:147)
3   libsystem_c.dylib             	0x00000001a23dbb80 abort + 116 (abort.c:118)
4   libsystem_c.dylib             	0x00000001a23dae78 __assert_rtn + 292 (assert.c:96)
5   xxxxxxxxxxx                  	0x0000000101b05ed0 dmMutex::TryLock(dmMutex::Mutex*) + 14704336 (mutex.cpp:60)
6   xxxxxxxxxxx                  	0x0000000101ac6168 dmHttpCache::Flush(dmHttpCache::Cache*) + 14442856 (http_cache.cpp:366)
7   xxxxxxxxxxx                  	0x0000000101a30538 dmHttpService::Loop(void*) + 13829432 (http_service.cpp:333)
8   xxxxxxxxxxx                  	0x0000000101ade5cc dmThread::ThreadStartProxy(void*) + 14542284 (thread.cpp:43)
9   libsystem_pthread.dylib       	0x00000001e4a57cb0 _pthread_start + 320 (pthread.c:881)
10  libsystem_pthread.dylib       	0x00000001e4a60778 thread_start + 8
```

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: engine/script/src/http_service.cpp [hunk 1]
```diff
@@ -395,28 +393,15 @@ namespace dmHttpService
...
-
             dmThread::Thread t = dmThread::New(&Loop, THREAD_STACK_SIZE, worker, "http");
-// Detach the thread on iOS
-// DNS lookups on iOS are using getaddrinfo which may block for an undefined
-// amount of time. We do not wish to wait for the thread during shutdown
-#if defined(DM_PLATFORM_IOS)
+
+            // Detach the thread
+            // DNS lookups using dmSocket::GetHostByName are using getaddrinfo which may block for an undefined
+            // amount of time. We do not wish to wait for the thread during shutdown
```
Comment (JCash) [inline engine/script/src/http_service.cpp:402]: Now we _always_ detach the thread.

@britzl Do you remember why we detach directly after creation, and not at shutdown?

Comment (britzl) [reply engine/script/src/http_service.cpp:402]: No. The documentation doesn't really say that you should use detach() at a specific point in a threads lifetime. You can create threads in a detached state or you can wait until shutdown, when you normally would call join().

**Review by britzl — DISMISSED** [review]

**Review by Jhonnyg — DISMISSED** [review]
Nice! 👍

File: engine/script/src/http_service.cpp [hunk 2]
```diff
@@ -222,10 +222,15 @@ namespace dmHttpService
...
             params.m_HttpCache = worker->m_Service->m_HttpCache;
-            params.m_DNSChannel = worker->m_DNSChannel;
             params.m_RequestTimeout = request->m_Timeout;
 
-            worker->m_Client = dmHttpClient::New(&params, url.m_Hostname, url.m_Port, strcmp(url.m_Scheme, "https") == 0);
+//dmLogInfo("NEW REQUEST enter  thread: %p  run: %d  time: %llu\n", worker->m_Thread, worker->m_Run? 1 : 0, dmTime::GetTime());
+
+            worker->m_Client = dmHttpClient::New(&params, url.m_Hostname, url.m_Port, strcmp(url.m_Scheme, "https") == 0, &worker->m_Canceled);
+
+//dmLogInfo("NEW REQUEST: client: %p  thread: %p  run: %d  time: %llu\n", worker->m_Client, worker->m_Thread, worker->m_Run? 1 : 0, dmTime::GetTime());
```
Comment (JCash) [inline engine/script/src/http_service.cpp:231]: removed

File: engine/script/src/http_service.cpp [hunk 3]
```diff
@@ -436,26 +429,42 @@ namespace dmHttpService
...
             url.m_Socket = worker->m_Socket;
-            dmDNS::StopChannel(http_service->m_Workers[i]->m_DNSChannel);
             dmMessage::Post(0, &url, 0, 0, (uintptr_t) dmHttpDDF::StopHttp::m_DDFDescriptor, 0, 0, 0);
-// On iOS we detach() the thread on creation so that we don't have to wait for
-// it during shutdown (see above for reason why)
-#if !defined(DM_PLATFORM_IOS)
-            dmThread::Join(worker->m_Thread);
-#endif
+
+            worker->m_Canceled = 1;
```
Comment (JCash) [inline engine/script/src/http_service.cpp:445]: A pointer to this flag is given to each new dmHttp::Client(), and allows it to cancel before a timeout occurs.

File: engine/script/src/http_service.cpp [hunk 4]
```diff
@@ -436,26 +429,42 @@ namespace dmHttpService
...
+        for (uint32_t i = 0; i < http_service->m_Workers.Size(); ++i)
+        {
+            dmHttpService::Worker* worker = http_service->m_Workers[i];
+
+            // DNS lookups using dmSocket::GetHostByName are using getaddrinfo which may block for an undefined
+            // amount of time. We do not wish to wait for the thread during shutdown
+            // We now use detach() on the thread on creation so that we don't have to wait for the thread
+            if (worker->m_Thread)
+            {
+                dmThread::Join(worker->m_Thread);
```
Comment (JCash) [inline engine/script/src/http_service.cpp:457]: We're back to joining!

**Pull request merged**